pkg/alertmanager/amcfg.go

```diff
@@ -138,20 +137,26 @@ func convertRoute(in *monitoringv1alpha1.Route, crKey types.NamespacedName, firs
 		delete(matchRE, "namespace")
 	}
 
-	// Set to nil if empty so that it doesn't show up in resulting yaml
+	// Set to nil if empty so that it doesn't show up in the resulting yaml.
 	if len(match) == 0 {
 		match = nil
 	}
-	// Set to nil if empty so that it doesn't show up in resulting yaml
+	// Set to nil if empty so that it doesn't show up in the resulting yaml.
 	if len(matchRE) == 0 {
 		matchRE = nil
 	}
 
 	var routes []*route
 	if len(in.Routes) > 0 {
-		routes := make([]*route, len(in.Routes))
-		for i := range in.Routes {
-			routes[i] = convertRoute(&in.Routes[i], crKey, false)
+		routes = make([]*route, len(in.Routes))
+		children, err := in.ChildRoutes()
+		if err != nil {
+			// The controller has already checked that Routes are valid. If an
+			// error is returned here, it's a bug.
+			panic(err)
```

Comment:
Can you clarify why we panic when we actually say above that error is returned here?

Reply:
In the comment, I meant to say that we shouldn't get any error here because child routes have already been checked in `checkAlertmanagerRoutes()` which gets called before generating the Alertmanager configuration.

https://github.com/prometheus-operator/prometheus-operator/pull/3703/files#diff-74002036bea7ef8f78c7623abba8d63a3e2578be73fb2dc8bcca154df57ce362R1087-R1103

Reply:
Makes sense now, thanks!